examples/tree_2d_dgsem/elixir_kpp.jl

```diff
@@ -0,0 +1,142 @@
+shock_indicator = IndicatorHennemannGassner(equation, basis,
+                                            alpha_max=0.5,
+                                            alpha_min=0.001,
+                                            alpha_smooth=true,
+                                            variable=first)
+volume_integral = VolumeIntegralShockCapturingHG(shock_indicator;
+                                                 volume_flux_dg=volume_flux,
+                                                 volume_flux_fv=surface_flux)
+solver = DGSEM(polydeg=polydeg, surface_flux=surface_flux, volume_integral=volume_integral)
+
+###############################################################################
+# Set up the tree mesh (initially a Cartesian grid of [-2,2]^2)
+coordinates_min = (-2.0, -2.0)
+coordinates_max = ( 2.0,  2.0)
+
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=6,
+                periodicity=true,
+                n_cells_max=500_000)
+
+###############################################################################
+# Create the semi discretization object
+semi = SemidiscretizationHyperbolic(mesh, equation, initial_condition_kpp, solver)
+
+###############################################################################
+# Set up adaptive mesh refinement
+amr_indicator = IndicatorHennemannGassner(semi,
+                                          alpha_max=1.0,
+                                          alpha_min=0.0001,
+                                          alpha_smooth=false,
+                                          variable=first)
+
+amr_controller = ControllerThreeLevelCombined(semi, amr_indicator, shock_indicator,
+                                              base_level=2,
+                                              med_level=0, med_threshold=0.0003,
+                                              max_level=8, max_threshold=0.003,
+                                              max_threshold_secondary=shock_indicator.alpha_max)
+
+amr_callback = AMRCallback(semi, amr_controller,
+                           interval=1,
+                           adapt_initial_condition=true,
+                           adapt_initial_condition_only_refine=true)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan)
+
+summary_callback = SummaryCallback()
+
+analysis_callback = AnalysisCallback(semi, interval=200)
+
+alive_callback = AliveCallback(analysis_interval=200)
+
+save_solution = SaveSolutionCallback(interval=200,
+                                     save_initial_solution=true,
+                                     save_final_solution=true,
+                                     solution_variables=cons2cons)
+
+callbacks = CallbackSet(summary_callback,
+                        analysis_callback, alive_callback,
+                        amr_callback, save_solution)
```

Comment:
```suggestion
                        save_solution, amr_callback)
```
The AMR callback should come after the `save_solution` callback since it technically already belongs to the next time step, compare also
https://github.com/trixi-framework/Trixi.jl/blob/a58e1a6d92dac6d66a891e36cc046ab57ac339df/src/callbacks_step/callbacks_step.jl#L44-L46